meeteval/wer/preprocess.py

```diff
@@ -0,0 +1,136 @@
+import itertools
+
+import meeteval
+from meeteval.io import SegLST
+from meeteval.io.pbjson import zip_strict
+from meeteval.io.seglst import seglst_map
+from meeteval.wer.wer.utils import check_single_filename
+
+
+@seglst_map()
+def split_words(
+        d: 'SegLST',
+        *,
+        keys=('words',)
+):
+    """
+    Splits segments into words and copies all other entries.
+
+    Uses a word-level timing strategy to convert the segment-level timings
+    into word-level timings. If no strategy is given, the timings are
+    copied as they are.
+
+    >>> split_words(SegLST([{'words': 'a b c'}]))
+    SegLST(segments=[{'words': 'a'}, {'words': 'b'}, {'words': 'c'}])
+
+    >>> split_words(SegLST([{'words': 'a b c', 'start_time': 0, 'end_time': 1}]))
+    SegLST(segments=[{'words': 'a', 'start_time': 0, 'end_time': 1}, {'words': 'b', 'start_time': 0, 'end_time': 1}, {'words': 'c', 'start_time': 0, 'end_time': 1}])
+
+    >>> split_words(SegLST([{'words': 'a b c', 'word_timings': [(0, 1), (1, 2), (2, 3)]}]), keys=('words', 'word_timings'))
+    SegLST(segments=[{'words': 'a', 'word_timings': (0, 1)}, {'words': 'b', 'word_timings': (1, 2)}, {'words': 'c', 'word_timings': (2, 3)}])
+    """
+    def split_entry(x):
+        if isinstance(x, str):
+            return x.split() or ['']
+        elif isinstance(x, (list, tuple)):
+            return x
+        else:
+            raise TypeError(x)
+
+    return d.flatmap(
+        lambda s: [
+            {**s, **dict(zip_strict(keys, split))}
+            for split in zip_strict(*(split_entry(s[key]) for key in keys))
+        ])
+
+
+def words_to_int(*d: SegLST):
+    """
+    Converts all words to ints. The mapping is created by iterating over all
+    words in d and assigning an integer to each word.
+
+    >>> words_to_int(SegLST([{'words': 'a b c'}]))
+    [SegLST(segments=[{'words': 1}])]
+
+    >>> words_to_int(SegLST([{'words': 'a'}, {'words': 'b'}]), SegLST([{'words': 'c'}, {'words': 'a'}]))
+    [SegLST(segments=[{'words': 4}, {'words': 2}]), SegLST(segments=[{'words': 3}, {'words': 4}])]
+
+    TODO: use cython code for speedup
+    TODO: unify everything. This stuff is done in multiple places in the code base.
+    """
+    # Convert into integer representation to save some computation later.
+    # `'words'` contains a single word only.
+    sym2int = {v: i for i, v in enumerate([
```

Comment:
Suggestion:
Replace this code with for `sym2int` with the following:
```
    sym2int = collections.defaultdict(itertools.count().__next__)
    _ = sym2int['']
```

It is slightly faster, the numbers are more natural, and the numbers stay smaller
and avoid potential overhead or int32 issues.

Reply:
Thanks, I forgot the set before enumerate. Your solution is shorter and cleaner!